testplan/runnable/base.py

```diff
@@ -669,30 +680,35 @@ def _merge_reports(self, test_report_lookup):
         merge_result = True
 
         for uid, reports in test_report_lookup.items():
-            count = reports[0][1].part[1]  # How many parts scheduled
+            parts = [(rep.part[0], rep.part[1]) for _, rep in reports]
```

Comment:
for the validity check this part of the code is traversing the reports/parts list at least six times, I admint that the error message is more specific than it was in previous implementation, but does it worth it?. It might be able to done in one for loop, collecting information, and the some iff making validation, if this detail is needded.

Reply:
Yes, I'll make a little change to it, making the purpose of code clearer, having less loop.